### PR TITLE
New feature: dataset custom schema

### PR DESCRIPTION
**Pull request by borism**
As per https://docs.google.com/spreadsheets/d/1D2UyTfHANXO5jO_RkBj5d6de5prkrpSe3KgYolBwTyk/edit#gid=1697736576

### CODE REVIEW

File: ckanext/honduras/dataset.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  "about_url": "http://github.com/datopian/ckanext-honduras",
+  "dataset_fields": [
+    {
+      "field_name": "title",
+      "label": "Nombre del Dataset",
+      "required": true,
+      "preset": "title"
+    },
+    {
+      "field_name": "description",
```
Comment (amercader) [inline ckanext/honduras/dataset.json]: Use `notes` here, as this is the default CKAN field for this

File: ckanext/honduras/dataset.json [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    },
+    {
+      "field_name": "identifier",
+      "label": "Identificador del Dataset",
+      "required": true
+    },
+    {
+      "field_name": "theme",
+      "label": "Temática del dataset",
+      "required": true
```
Comment (amercader) [inline ckanext/honduras/dataset.json]: this should be a [`select` preset](https://github.com/ckan/ckanext-scheming/#choices). The `choices` are in the "VARIABLES" sheet of the spreadsheet (Column "Categoría de Información"). Use the values on the spreadsheet as labels, and for values the lower-case, hyphenated form of them (also replacing accents like í -> i)

File: ckanext/honduras/dataset.json [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    },
+    {
+      "field_name": "keyword",
```
Comment (amercader) [inline ckanext/honduras/dataset.json]: Use the core `tags` field for this, as described [here](https://github.com/ckan/ckanext-scheming/blob/master/ckanext/scheming/ckan_dataset.json#L25) but changing the label.

File: ckanext/honduras/dataset.json [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      "label": "Palabras clave",
+      "required": true
+    },
+    {
+      "field_name": "issued",
+      "label": "Fecha de Creación del Dataset"
```
Comment (amercader) [inline ckanext/honduras/dataset.json:138]: Use the `date` preset here

File: ckanext/honduras/dataset.json [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    },
+    {
+      "field_name": "modified",
```
Comment (amercader) [inline ckanext/honduras/dataset.json]: Use the `date` preset here

File: ckanext/honduras/dataset.json [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      "label": "Fecha de última actualización del Dataset",
+      "required": true
+    },
+    {
+      "field_name": "accrualPeriodicity",
```
Comment (amercader) [inline ckanext/honduras/dataset.json:146]: this should be a [`select` preset](https://github.com/ckan/ckanext-scheming/#choices). The `choices` are in the "VARIABLES" sheet of the spreadsheet (Column "Frecuencia de actualización de la información"). Use the values on the spreadsheet as labels, and for values the lower-case, hyphenated form of them (also replacing accents like í -> i)

File: ckanext/honduras/dataset.json [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    {
+      "field_name": "describedBy",
+      "label": "Documentación asociada al Dataset (Diccionario de Datos o URL a modelo de datos)"
+    },
+    {
+      "field_name": "references",
+      "label": "Normativa asociada al Dataset (URL)"
+    },
+    {
+      "field_name": "license",
```
Comment (amercader) [inline ckanext/honduras/dataset.json]: Copy [this field](https://github.com/ckan/ckanext-scheming/blob/master/ckanext/scheming/ckan_dataset.json#L32) definition

File: ckanext/honduras/dataset.json [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+      "field_name": "identifier",
+      "label": "Identificador de la Distribución"
+    },
+    {
+      "field_name": "name",
+      "label": "Nombre de la Distribución",
+      "required": true
+    },
+    {
+      "field_name": "decription",
```
Comment (amercader) [inline ckanext/honduras/dataset.json]: typo `description`

File: ckanext/honduras/dataset.json [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+      "label": "Descripción de la Distribución",
+      "required": true
+    },
+    {
+      "field_name": "mediaType",
```
Comment (amercader) [inline ckanext/honduras/dataset.json]: Let's use the [default `format`](https://github.com/ckan/ckanext-scheming/blob/master/ckanext/scheming/ckan_dataset.json#L101) one here

File: ckanext/honduras/dataset.json [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    {
+      "field_name": "byteSize",
+      "label": "Tamaño de la Distribución"
+    },
+    {
+      "field_name": "describedBy",
+      "label": "Documentación asociada a la Distribución (Diccionario de Datos o URL a modelo de datos)"
+    },
+    {
+      "field_name": "accessURL",
```
Comment (amercader) [inline ckanext/honduras/dataset.json]: Can you replace `accessURL` and `downloadURL` with the [standard `url` one](https://github.com/ckan/ckanext-scheming/blob/master/ckanext/scheming/ckan_dataset.json#L86) and move it to the top of the resource fields?

**Review by amercader — CHANGES_REQUESTED** [review]
Nice work @borism , I added some comments to make it more aligned with the standard CKAN schema

**Comment by sivang** [comment]
hey, I saw this update and was wondering, shouldn't schema files be placed under a `ckanext/honduras/schema/` if one uses `ckanext-scheming` which I assume you do here or perhaps this is not critical?

**Comment by borism** [comment]
> hey, I saw this update and was wondering, shouldn't schema files be placed under a `ckanext/honduras/schema/` if one uses `ckanext-scheming` which I assume you do here or perhaps this is not critical?

it works like this when I add `scheming.dataset_schemas = ckanext.honduras:dataset.json` to ckan config. perhaps there's a better place for storing this, I'm open to suggestions.

**Comment by amercader** [comment]
Great work @borism. I added some final tweaks on 941aa86 directly so you can focus on the importer.

**Comment by amercader** [comment]
@sivang You can put the json schemas wherever you want in the extension as long as you reference it correctly in the config option.. /schemas/ is more of a convention when there's more than one.

**Pull request merged**